exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/AvgWidthMergedStatistic.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.drill.exec.vector.NullableFloat8Vector;
+import org.apache.drill.exec.vector.ValueVector;
+import org.apache.drill.exec.vector.complex.MapVector;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AvgWidthMergedStatistic extends AbstractMergedStatistic {
+
+  private String name;
+  private String inputName;
+  private boolean configureComplete = false;
+  private boolean mergeComplete = false;
+  private Map<String, ValueHolder> sumHolder;
+  MergedStatistic types, nonNullStatCounts, statCounts;
+
+  public AvgWidthMergedStatistic (String name, String inputName) {
+    this.name = name;
+    this.inputName = inputName;
+    this.sumHolder = new HashMap<>();
+    types = nonNullStatCounts = statCounts = null;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getInput() {
+    return inputName;
+  }
+
+  @Override
+  public void merge(ValueVector input) {
+    // Check the input is a Map Vector
+    assert (input.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    MapVector inputMap = (MapVector) input;
+    for (ValueVector vv : inputMap) {
+      String colName = vv.getField().getLastName();
+      NullableFloat8Holder colSumHolder;
+      if (sumHolder.get(colName) != null) {
+        colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      } else {
+        colSumHolder = new NullableFloat8Holder();
+        sumHolder.put(colName, colSumHolder);
+      }
+      Object val = vv.getAccessor().getObject(0);
+      if (val != null) {
+        colSumHolder.value += (double) val;
+        colSumHolder.isSet = 1;
+      }
+    }
+  }
+
+  @Override
+  public Object getStat(String colName) {
+      if (mergeComplete != true) {
+        throw new IllegalStateException(
+            String.format("Statistic `%s` has not completed merging statistics", name));
+      }
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      return (long) (colSumHolder.value/ getRowCount(colName));
+    }
+
+  @Override
+  public void setOutput(ValueVector output) {
+    // Check the input is a Map Vector
+    assert (output.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    // Dependencies have been configured correctly
+    assert (configureComplete == true);
+    MapVector outputMap = (MapVector) output;
+
+    for (ValueVector outMapCol : outputMap) {
+      String colName = outMapCol.getField().getLastName();
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      NullableFloat8Vector vv = (NullableFloat8Vector) outMapCol;
+      vv.allocateNewSafe();
+      vv.getMutator().setSafe(0, (colSumHolder.value / getRowCount(colName)));
+    }
+    mergeComplete = true;
+  }
+
+  @Override
+  public void configure(Object configurations) {
+    List<MergedStatistic> statistics = (List<MergedStatistic>) configurations;
```

Comment:
```
public void configure(List<MergedStatistic> configurations) {
```

Please don't use generic Object and casting unless absolutely necessary. (Sometimes it is. This doesn't appear to be one of those times...)